packages/langium/src/utils/errors.ts

```diff
@@ -0,0 +1,17 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { CstNode } from '..';
+
+export class ErrorWithLocation extends Error {
+    constructor(node: CstNode|undefined, message: string) {
+        super(`${message} at ${node?.range.start.line || 0}:${node?.range.start.character || 0}`);
+    }
+}
+
+export function assertUnreachable(_: never): never {
+    throw new Error('Error! Please fulfill the logic also for the other types in the inheritence hierarchy.');
```

Comment:
Minor: The error message is a bit too specific for my taste. This error can happen anytime we get an unexpected/unhandled value in an `if-else/switch` block, so I imagine something like this:

```suggestion
    throw new Error('Error! The input value was not handled.');
```

Reply:
`assertUnreachable` is a compile-time trick and will lead to an error, when the parameter is not the empty type. So actually we never reach this point.

Reply:
I see, alright then.